src/sql/workbench/contrib/query/common/queryResultsWriterStatus.ts

```diff
@@ -0,0 +1,38 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Disposable } from 'vs/base/common/lifecycle';
+
+export enum QueryResultsWriterMode {
+	ToGrid,
+	ToFile,
+}
+
+export class QueryResultsWriterStatus extends Disposable {
+	public static instance: QueryResultsWriterStatus = undefined;
+
+	private writerMode: QueryResultsWriterMode;
+
+	constructor(mode: QueryResultsWriterMode = QueryResultsWriterMode.ToGrid) {
+		super();
+		this.writerMode = mode;
+	}
+
+	public set mode(mode: QueryResultsWriterMode) {
+		this.writerMode = mode;
+	}
+
+	public get mode() {
+		return this.writerMode;
+	}
+
+	public isWritingToGrid(): boolean {
```

Comment:
Seems easier to just expose the writerMode directly and let callers have logic based on that